File: Be-siukat-new/siukat-service-dev_snmptn_2022/models/log_kendaraan.js
var Sequelize = require('sequelize')
var db = require('../config/database')

const LogKendaraan = db.define('log_kendaraan', {
  id_log_kendaraan:{
  	type: Sequelize.INTEGER,
  	primaryKey: true
  },
  no_peserta: {
    type: Sequelize.STRING
  },
  status_motor: {
    type: Sequelize.ENUM('memiliki', 'tidak_memiliki')
  },
  jumlah_motor: {
    type: Sequelize.INTEGER
  },
  pajak_motor: {
    type: Sequelize.INTEGER
  },
  scan_motor: {
    type: Sequelize.STRING
  },
  status_mobil: {
    type: Sequelize.ENUM('memiliki', 'tidak_memiliki')
  },
  jumlah_mobil: {
    type: Sequelize.INTEGER
  },
  pajak_mobil: {
    type: Sequelize.INTEGER
  },
  scan_mobil: {
    type: Sequelize.STRING
  },
  atribut: {
    type: Sequelize.ENUM('original', 'sanggah')
  },
  executor: {
    type: Sequelize.STRING
  },
  timestamp: {
    type: Sequelize.DATE
  }
},
 {
  // don't add the timestamp attributes (updatedAt, createdAt)
  timestamps: false,

  // don't delete database entries but set the newly added attribute deletedAt
  // to the current date (when deletion was done). paranoid will only work if
  // timestamps are enabled
  paranoid: true,

  // don't use camelcase for automatically added attributes but underscore style
  // so updatedAt will be updated_at
  underscored: true,

  // disable the modification of tablenames; By default, sequelize will automatically
  // transform all passed model names (first parameter of define) into plural.
  // if you don't want that, set the following
  freezeTableName: true,

  // define the table's name
  tableName: 'log_kendaraan'
}
);
module.exports = LogKendaraan

File: BackEnd-Siukat/models/log_kendaraan.go
package models

import "time"

type LogKendaraan struct {
	IDLogKendaraan int        `gorm:"primaryKey;column:id_log_kendaraan" json:"id_log_kendaraan"`
	NoPeserta      string     `gorm:"column:no_peserta;type:varchar(255)" json:"no_peserta"`
	StatusMotor    string     `gorm:"column:status_motor;type:enum('memiliki','tidak_memiliki')" json:"status_motor"`
	JumlahMotor    int        `gorm:"column:jumlah_motor" json:"jumlah_motor"`
	PajakMotor     int        `gorm:"column:pajak_motor" json:"pajak_motor"`
	ScanMotor      string     `gorm:"column:scan_motor;type:varchar(255)" json:"scan_motor"`
	StatusMobil    string     `gorm:"column:status_mobil;type:enum('memiliki','tidak_memiliki')" json:"status_mobil"`
	JumlahMobil    int        `gorm:"column:jumlah_mobil" json:"jumlah_mobil"`
	PajakMobil     int        `gorm:"column:pajak_mobil" json:"pajak_mobil"`
	ScanMobil      string     `gorm:"column:scan_mobil;type:varchar(255)" json:"scan_mobil"`
	Atribut        string     `gorm:"column:atribut;type:enum('original','sanggah')" json:"atribut"`
	Executor       string     `gorm:"column:executor;type:varchar(255)" json:"executor"`
	Timestamp      *time.Time `gorm:"column:timestamp;type:datetime" json:"timestamp"`
}

func (LogKendaraan) TableName() string {
	return "log_kendaraan"
}
